### PR TITLE
feat(ensemble): skill-based model weighting (beat-naive-or-zero)

### PR DESCRIPTION
**Pull request by saikrishna-dev**
## Summary

- Replaces `1/MAE` historical weights with `max(0, 1 − model_mae/naive_mae)` — weights proportional to how much each model beats naive persistence
- Adds `ForecastStore.query_naive_mae()` — computes persistence MAE from existing `forecast_record` + `price_outcome` InfluxDB data (no new measurements)
- Runs all 3 RL queries concurrently via `asyncio.gather` (minor latency improvement)
- Falls back to `1/MAE` for tickers with no naive baseline yet; falls back to equal weights when all models are ≤ naive

## Why

The previous `1/MAE` weighting can reward persistence-mimicry: a model that just predicts `last_price` each day gets a low MAE on random-walk assets and therefore a high weight. The scoreboard (PR #245) made this visible. This PR fixes it — a model must beat the trivial baseline to earn weight in the ensemble.

## API contract

No changes to request/response shapes. The `modelWeights` field in `/predict` continues to return `{prophet, sarima, rf}` fractions — they now reflect skill rather than raw inverse-MAE.

## Test plan

- [ ] `python -m pytest backend/tests/test_skill_weights.py -v` — 7 unit tests, all pure logic
- [ ] `python -m pytest backend/tests/ --tb=short` — full suite (132 tests)
- [ ] `ruff check backend/` — clean
- [ ] Verify `/predict AAPL` logs show `[RL] Skill-based weights` once the ticker has resolved forecast records; new tickers log `Inv-MAE (no naive baseline yet)` as expected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Enhanced ensemble forecast weighting with skill-based calibration that accounts for naive baseline performance metrics.
  * Improved concurrent metric retrieval for faster forecast calculations.

* **Tests**
  * Added comprehensive test coverage for ensemble weight calibration logic to ensure stability across edge cases.

### CODE REVIEW

File: backend/routers/predict.py [hunk 1]
```diff
@@ -1009,9 +1010,13 @@ async def _predict_inner(payload: PredictRequest) -> PredictionResponse:
...
     # ── RL: load historical model accuracy for weight calibration ────────────
+    # All three queries run concurrently — no inter-dependency.
     logger.debug(f"[RL] Querying historical model accuracy for {symbol} ...")
-    model_acc = await asyncio.to_thread(forecast_store.query_model_accuracy, symbol)
-    ensemble_mae = await asyncio.to_thread(forecast_store.query_ensemble_mae, symbol)
+    model_acc, ensemble_mae, naive_mae_val = await asyncio.gather(
+        asyncio.to_thread(forecast_store.query_model_accuracy, symbol),
+        asyncio.to_thread(forecast_store.query_ensemble_mae,   symbol),
+        asyncio.to_thread(forecast_store.query_naive_mae,      symbol),
+    )
```
[the remark this answers is not in the document]
Comment (saikrishna-dev) [reply backend/routers/predict.py:1027]: Fixed in commit 4fa2e3f — wrapped all three to_thread calls in a local `_rl_query` helper that applies `asyncio.wait_for(timeout=12.0)` and logs a WARNING + returns the safe default on any exception.

File: backend/services.py [hunk 2]
```diff
@@ -726,6 +726,30 @@ def query_price_outcomes(self, symbol: str, days: int = 15) -> Dict[object, floa
...
+                continue
+            try:
+                pred_date = pred_time.date()
+            except AttributeError:
+                continue
+            for d, close in outcomes_sorted:
+                if d > pred_date and close > 0:
+                    errors.append(abs(close - float(last_price)))
+                    break
+        return round(sum(errors) / len(errors), 4) if errors else None
```
[the remark this answers is not in the document]
Comment (saikrishna-dev) [reply backend/services.py]: Fixed in commit 4fa2e3f — wrapped the full body in try/except, added explicit `float()` casts with per-record AttributeError/TypeError/ValueError handling, and added an outer guard that logs WARNING and returns None on any unexpected failure.
